src/main/java/com/revature/globetrotters/controller/PostController.java

```diff
@@ -0,0 +1,101 @@
+package com.revature.globetrotters.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import com.revature.globetrotters.entity.Post;
+import com.revature.globetrotters.entity.PostComment;
+import com.revature.globetrotters.service.PostService;
+
+@Controller
+@CrossOrigin(origins = "http://localhost:5173")
+public class PostController {
+
+    @Autowired
+    private PostService postService;
+
+    @GetMapping("users/{userId}/posts")
+    public ResponseEntity<List<Post>> getAllPosts(@PathVariable int userId) {
+        return ResponseEntity.status(HttpStatus.OK).body(postService.findPostsByUserId(userId));
+    }
+
+    @PostMapping("post")
+    public ResponseEntity<Post> postPost(Post newPost) {
+        try {
+            Post post = postService.createPost(newPost);
+            return ResponseEntity.status(HttpStatus.OK).body(post);
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null);
+        }
+    }
+
+    @GetMapping("posts/{postId}")
+    public ResponseEntity<Post> getPostById(@PathVariable int postId) throws Exception {
+        return ResponseEntity.status(HttpStatus.OK).body(postService.findPostById(postId));
+    }
+
+    @DeleteMapping("posts/{postId}")
```

Comment:
We probably don't need to send a message back saying that we deleted the post. A response with an OK status code would indicate that the post was deleted and a BadRequest status code would indicate that a post with the given id doesn't exist.